### PR TITLE
Fix lts

### PR DESCRIPTION
**Pull request by sebastianbenz**
The new runtime parameter handling resulted in the LTS option no longer
working. Moving lts + ampRuntimeVersion in fetchRuntimeParameters.

This PR also improves logging and fixes a bug where the styles fallback
did not work.

### CODE REVIEW

**Comment by sebastianbenz** [comment]
@mdmower would be great if you could take a look.

File: packages/optimizer/lib/RuntimeHostHelper.js [hunk 1]
```diff
@@ -17,11 +17,12 @@
 
 const {AMP_CACHE_HOST, appendRuntimeVersion} = require('./AmpConstants.js');
 
-function calculateHost({ampUrlPrefix = AMP_CACHE_HOST, ampRuntimeVersion, lts, rtv = false}) {
-  if (lts && ampRuntimeVersion) {
-    throw new Error('lts flag is not compatible with runtime version parameter');
```
Comment (mdmower) [inline packages/optimizer/lib/RuntimeHostHelper.js:22]: I think this error is worth keeping, but `if (lts && ampRuntimeVersion)` should be replaced with `if (lts && rtv)`.

File: packages/optimizer/lib/fetchRuntimeParameters.js [hunk 2]
```diff
@@ -149,11 +155,17 @@ async function fetchAmpRuntimeVersion_(context) {
...
     version: await config.runtimeVersion.currentVersion({ampUrlPrefix, lts}),
     maxAge: MaxAge.create(AMP_RUNTIME_MAX_AGE).toObject(),
   };
-  if (versionKey) {
+  if (!ampRuntimeData.version && ampUrlPrefix !== AMP_CACHE_HOST) {
+    config.log.error(
+      `Could not download runtime version from ${ampUrlPrefix}. Falling back to ${AMP_CACHE_HOST}`
+    );
+    ampRuntimeData = await fetchLatestRuntimeData_({config, AMP_CACHE_HOST, lts}, versionKey);
+  } else if (versionKey) {
```
Comment (mdmower) [inline packages/optimizer/lib/fetchRuntimeParameters.js:168]: Do you want to store bad results? If not:
`} else if (ampRuntimeData.version && versionKey) {`

File: packages/optimizer/lib/RuntimeHostHelper.js (hunk 1 above)
Comment (sebastianbenz) [reply packages/optimizer/lib/RuntimeHostHelper.js:22]: Good point. Added.

File: packages/optimizer/lib/fetchRuntimeParameters.js (hunk 2 above)
Comment (sebastianbenz) [reply packages/optimizer/lib/fetchRuntimeParameters.js]: Done

**Pull request merged**